### PR TITLE
Documentation: Added Styling Information To CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by bedwardly-down**
After running into my own frustrations with styling and whatnot, I decided to add this for those that may not know where to look or how to even go about making the user interface code fit with the rest of what is already implemented.

### CODE REVIEW

**Comment by bedwardly-down** [comment]
It looks like the build failed because of an issue with the Travis build server and not because I made a code change.

**Comment by laurent22** [comment]
Thanks, it's good to clarify this. I feel the paragraph would be more suitable for the Coding Style section though, could you live it there please?

In fact we should probably have a separate document for all coding guidelines, as Contributing.md is more a general overview.

**Comment by bedwardly-down** [comment]
> In fact we should probably have a separate document for all coding guidelines, as Contributing.md is more a general overview.

Most definitely. I even posted something along those lines on the forums

https://discourse.joplinapp.org/t/contributing-md-suggestions/2317/10?u=bedwardly-down

**Comment by bedwardly-down** [comment]
Here's the section movement. I don't feel completely confident with starting a new document but could definitely either try or move this to it if one of you starts it or gives me some kind of guideline for it. Thanks

**Comment by tessus** [comment]
@bedwardly-down e.g. here's an [example](https://evermeet.cx/wiki/Coding_Conventions_(PHP)) of how a coding guidline could look like. Some sections are missing iike Classnames, Methods, ... but I think you get the idea.

**Comment by bedwardly-down** [comment]
> @bedwardly-down e.g. here's an [example](https://evermeet.cx/wiki/Coding_Conventions_(PHP)) of how a coding guidline could look like. Some sections are missing iike Classnames, Methods, ... but I think you get the idea.

Gotya. So this is more technical based than general knowledge based.

**Comment by tessus** [comment]
I'd say so. Someone who writes code already has some sort of technical understanding. Coding conventions are a technical subject.

**Comment by laurent22** [comment]
For the most part our coding conventions are enforced by eslint so we don't need a very detailed document.

Off the top of my head what we should mention at least are:

- Styling (what you've written)
- Usage of React Hooks for new components
- Usage of TypeScript for new file
- Make sure that eslint is running

As it is we can already merge your Contributing.md, but for the coding guidelines I think we discuss that in the forum first.

What would be good is to find an existing example or template and work from that.

**Comment by bedwardly-down** [comment]
Topic made: https://discourse.joplinapp.org/t/upon-laurents-request-lets-discuss-improve-documentation-and-styling-guidelines/5806

**Comment by PackElend** [comment]
I added my thought in the forum although they go beyond styling of the UI

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -46,6 +46,8 @@ Coding style is enforced by a pre-commit hook that runs eslint. This hook is ins
 
 For new React components, please use [React Hooks](https://reactjs.org/docs/hooks-intro.html). For new code in general, please use TypeScript (unless you are modifying a file that was originally in JavaScript).
 
+- For changes made to the Desktop client that affect the user interface, refer to `ElectronClient/app/theme.js` for all styling information. The goal is to create a consistent user interface to allow for easy navigation of Joplin's various features and improve the overall user experience.
+
```
Comment (laurent22) [inline CONTRIBUTING.md]: That's good to merge but please remove the `-` at the beginning of the line.

File: CODING-GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+## Coding Style
+
+
+- All functions, classes, components and properties must be clearly and concisely named so that upon inspection, one can quickly and easily discern what the purposes of each are and how they are associated with the project as a whole.
+- In order to further improve code readability, comments are welcome but need to be as short and precise as needed in order to assist but not distract from the rest of the code.
+
+## Desktop (ElectronClient) guidelines
+
+
+- For changes that affect the user interface, refer to `ElectronClient/app/theme.js` for all styling information. This is to allow for easy navigation of Joplin's various features and improve the overall user experience.
```
Comment (laurent22) [inline CODING-GUIDELINES.md]: Please remove this file from the commit. Coding guidelines would definitely be useful, but there's currently not much time to look into that due to GSoC.

**Comment by laurent22** [comment]
@bedwardly-down, please let me know if you're still interested in merging this PR after the requested changes have been made.

**Comment by bedwardly-down** [comment]
Sorry. Haven't been receiving any notifications for this PR and have been so caught up in the bug I'm working through with everyone else.

**Comment by laurent22** [comment]
No problem @bedwardly-down, and thanks for the updated PR!

**Pull request merged**